lib/plugins/aws/deployFunction/index.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const fs = require('fs');
+const path = require('path');
+const _ = require('lodash');
+const Zip = require('node-zip');
+const SDK = require('../');
+const validate = require('../lib/validate');
+
+class AwsDeployFunction {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = 'aws';
+    this.sdk = new SDK(serverless);
+
+    this.functionName =
+      `${this.serverless.service.service}-${this.options.stage}-${this.options.function}`;
+
+    Object.assign(this, validate);
+
+    this.hooks = {
+      'deploy:function:deploy': () => BbPromise.bind(this)
+        .then(this.validate)
+        .then(this.checkIfFunctionExistsInService)
+        .then(this.checkIfFunctionIsDeployed)
+        .then(this.zipFunction)
+        .then(this.deployFunction),
+    };
+  }
+
+  checkIfFunctionExistsInService() {
```

Comment:
do we really need this method wrapper just to check the function exists? We can just move it to one of the other methods. I guess that's pretty opinionated though :)